### PR TITLE
let httr build URL

### LINKED ISSUE #9: Problem with non-ASCII characters on Windows
I got this error with R 3.3.1 on Windows 10.

```r
WikipediR::page_content("ja", "wikipedia", page_name = "ポケモン")
#> Error in query(url, "pcontent", clean_response, ...) : 
#>   The API returned an error: invalidtitle - Bad title "<U+0192>|<U+0192>P<U+0192><U+201A><U+0192>“"
```

If I convert the parameter for `page_name` into UTF-8 explicitly, it works.

```r
txt <- WikipediR::page_content("ja", "wikipedia", page_name = iconv("ポケモン", to = "UTF-8"))
```

I guess it would be better to pass all arguments to `query` param of `GET()` since `httr` will take care of encodings.

```r
library(httr)

res <- GET("http://en.wikipedia.org/w/api.php",
    query = list(
      format = "json",
      action = "parse",
      prop   = "text|revid",
      page   = "ポケモン")
)
content(res)
#> $parse
#> $parse$title
#> [1] "ポケモン"
#> 
#> $parse$pageid
#> [1] 3211650
#> 
#> $parse$revid
#> [1] 456775067
#> ...
```

### PR DESCRIPTION
**Pull request by yutannihilation**
fixes #9

Note that there is one breaking change; `query()` gets `query_param` arg. Those who use `query()` with unnamed argument may suffer from this change (but I bet there are very few...)

### CODE REVIEW

**Comment by okeyes-r7** [comment]
Well, query isn't exported (I don't think?) so shouldn't be an issue.

Again, will review this evening (well, test this evening, I can go through code now)

File: R/query.R [hunk 1]
```diff
@@ -11,17 +11,19 @@
...
 #'by out_class, or not.
 #'
+#'@param query_param query parameters
+#'
 #'@param ... further arguments to httr's GET.
 #'@export
-query <- function(url, out_class, clean_response = FALSE, ...){
+query <- function(url, out_class, clean_response = FALSE, query_param = list(), ...){
+  # Common query parameters
+  if(is.null(query_param$format)) query_param$format <- "json"
```
Comment (okeyes-r7) [inline R/query.R]: Please use a multiline if statement, here.

File: man/query.Rd [hunk 2]
```diff
@@ -15,6 +15,8 @@ WikidataR to indicate what response-cleaning method should be applied.}
 \item{clean_response}{whether to clean the response, using the method assigned
 by out_class, or not.}
 
+\item{query_param}{query parameters}
```
Comment (okeyes-r7) [inline man/query.Rd]: Does this cause choking in the WikidataR package?

**Comment by yutannihilation** [comment]
Thanks for comments!

> Well, query isn't exported

Oh, but query is (unintentionally?) exported...: https://github.com/Ironholds/WikipediR/blob/8556b104747cb560dd64b7e03ea0c13a0d82cf34/R/query.R#L15

According to the CRAN page, WikipediR is imported only by WikidataR. It seems WikidataR used to use `query()`, but I couldn't find any query() is used actually: https://github.com/Ironholds/WikidataR/blob/c1b96f9c38969ebc0454d203a13820052ae7ccfe/R/WikidataR.R#L11

**Comment by Ironholds** [comment]
...submit a patch to WikidataR, I'll merge both? ;)

**Comment by yutannihilation** [comment]
Thanks for suggestion, I will :)

But this will break WikidataR until the next versions of WikipediR and WikidataR both will be available on CRAN. So I will add a temporal treatment.

I've checked `WikidataR::find_item()` with this branch version of WikipediR. Then, I got this error:

```r
library(WikidataR)
adams_items <- find_item("Douglas Adams")
#> Error in curl::curl_fetch_memory(url, handle = handle) : 
#>   Couldn't connect to server
```

This is because the `GET()`ing [without `"http://"` scheme](https://github.com/Ironholds/WikidataR/blob/cb36f544d6a7b702528ddb4fb2de0f147524ed2d/R/utils.R#L34) and with `query` param will return errors. So `query()` should add `http://` to `url` if there is no `http://`.

```r
# GET() without "http://" prefix and without query arg is OK.
httr::GET(
  "wikidata.org/w/api.php?action=wbsearchentities&format=json&type=item&language=en&limit=10&search=Douglas Adams"
)

# GET() without "http://" prefix and with query arg returns an error.
httr::GET(
  "wikidata.org/w/api.php?action=wbsearchentities&format=json&type=item&language=en&limit=10&search=Douglas Adams",
  query = list(format = "json")
)

# GET() with "http://" prefix and with query arg is OK.
httr::GET(
  "http://wikidata.org/w/api.php?action=wbsearchentities&format=json&type=item&language=en&limit=10&search=Douglas Adams",
  query = list(format = "json")
)
```

**Comment by okeyes-r7** [comment]
Well, it will break the GitHub versions. I'm going to push them up to CRAN shortly.




Oliver Keyes | Senior Data Scientist

Email: Oliver_Keyes@rapid7.com<mailto:first_last@rapid7.com>
Twitter: @quominus


________________________________

**Comment by yutannihilation** [comment]
I've submitted the PR to WikidataR!

> Well, it will break the GitHub versions. I'm going to push them up to CRAN shortly.

Nice. Then, should I remove this commit that I added for the compatibility with the current version of WikidataR? https://github.com/Ironholds/WikipediR/pull/10/commits/eeccce8018c0d2243177003dc056ef9d3251e307

**Pull request merged**